### PR TITLE
Snow 1417552 alias snowcli spcs object commands

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add aliases for `snow object list/drop/describe` commands for `compute-pool`, `image-repository` and `service` objects

### CODE REVIEW

File: tests/__snapshots__/test_help_messages.ambr [hunk 1]
```diff
@@ -3504,6 +3503,215 @@
...
+   Usage: default spcs compute-pool list [OPTIONS]                                
+                                                                                  
+   Lists all available compute pools.                                             
+                                                                                  
+  ╭─ Options ────────────────────────────────────────────────────────────────────╮
+  │ --like  -l      TEXT            SQL LIKE pattern for filtering objects by    │
+  │                                 name. For example, `list --like "my%"` lists │
+  │                                 all compute pools that begin with “my”..     │
+  │                                 [default: %%]                                │
+  │ --in            <TEXT TEXT>...  Specifies the scope of this command using    │
```
Comment (sfc-gh-yiwang) [inline tests/__snapshots__/test_help_messages.ambr:3656]: The in scope option doesn't work for compute pool

File: src/snowflake/cli/plugins/object/command_aliases.py [hunk 2]
```diff
@@ -17,35 +18,51 @@ def add_object_command_aliases(
...
-
-    @app.command("drop", requires_connection=True)
-    def drop_cmd(name: str = name_argument, **options):
-        drop(
-            object_type=object_type.value.cli_name,
-            object_name=name,
+    if "list" not in ommit_commands:
+        if not like_option or not scope_option:
+            raise ClickException(
+                '[like_option] and [scope_option] parameters have to be defined for "list" command'
```
[the remark this answers is not in the document]
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/object/command_aliases.py]: Good point! Fixed

File: src/snowflake/cli/plugins/spcs/compute_pool/commands.py [hunk 3]
```diff
@@ -71,6 +80,16 @@ def _compute_pool_name_callback(name: str) -> str:
...
 _COMMENT_HELP = "Comment for the compute pool."
 
+add_object_command_aliases(
+    app=app,
+    object_type=ObjectType.COMPUTE_POOL,
+    name_argument=ComputePoolNameArgument,
+    like_option=like_option(
+        help_example='`list --like "my%"` lists all compute pools that begin with “my”.'
+    ),
+    scope_option=scope_option(help_example="`list --in database my_db`"),
```
[the remark this answers is not in the document]
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/spcs/compute_pool/commands.py]: Fixed

File: tests/__snapshots__/test_help_messages.ambr (hunk 1 above)
Comment (sfc-gh-pczajka) [reply tests/__snapshots__/test_help_messages.ambr]: fixed

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -10,6 +10,9 @@
    * `snow git` for git repository stages
    * `snow streamlit` for streamlit apps
    * `snow snowpark` for procedures and functions
+   * `snow spcs compute-pool` for compute pools
+   * `snow spcs image-repository` for image repositories
+   * `snow spcs services` for services
```
[the remark this answers is not in the document]
Comment (sfc-gh-mraba) [reply RELEASE-NOTES.md:15]: `compute-pool` / `image-repository` are singular. IMO it should be singular `service`

File: src/snowflake/cli/plugins/object/command_aliases.py [hunk 5]
```diff
@@ -17,35 +19,62 @@ def add_object_command_aliases(
     app: SnowTyper,
     object_type: ObjectType,
     name_argument: typer.Argument,
-    like_option: typer.Option,
-    scope_option: typer.Option,
+    like_option: Optional[typer.Option],
+    scope_option: Optional[typer.Option],
+    ommit_commands: List[str] = [],
```
Comment (sfc-gh-yiwang) [inline src/snowflake/cli/plugins/object/command_aliases.py:24]: typo here? ommit -> omitted

**Review by sfc-gh-yiwang — APPROVED** [review]

File: src/snowflake/cli/plugins/object/command_aliases.py (hunk 5 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/object/command_aliases.py:24]: Intended: "ommit these while creating aliases"

**Review by sfc-gh-mraba — APPROVED** [review]

File: src/snowflake/cli/plugins/object/command_aliases.py (hunk 5 above)
Comment (sfc-gh-yiwang) [reply src/snowflake/cli/plugins/object/command_aliases.py:24]: It only contains 1 'm'.  ommit -> omit

**Pull request merged**